cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterRebalanceAssemblyOperator.java

```diff
@@ -0,0 +1,804 @@
+                                if (getRebalanceAnnotation(freshClusterRebalance) == RebalanceAnnotation.stop) {
+                                    log.debug("{}: Stopping current Cruise Control proposal request timer", reconciliation);
+                                    vertx.cancelTimer(t);
+                                    p.complete(new KafkaClusterRebalanceStatusBuilder()
+                                            .withSessionId(null)
+                                            .addNewCondition().withNewType(State.Stopped.toString()).endCondition().build());
+                                } else {
+                                    requestRebalance(reconciliation, host, apiClient, true, rebalanceOptionsBuilder,
+                                            freshClusterRebalance.getStatus().getSessionId()).setHandler(rebalanceResult -> {
+                                                if (rebalanceResult.succeeded()) {
+                                                    // If the returned status has an optimization result then the rebalance proposal
+                                                    // is ready, so stop the polling
+                                                    if (rebalanceResult.result().getOptimizationResult() != null &&
+                                                            !rebalanceResult.result().getOptimizationResult().isEmpty()) {
+                                                        vertx.cancelTimer(t);
+                                                        log.debug("Optimization proposal ready");
+                                                        p.complete(rebalanceResult.result());
+                                                    } else {
+                                                        log.debug("Waiting for optimization proposal to be ready");
+                                                    }
+                                                    // The rebalance proposal is still not ready yet, keep the timer for polling
+                                                } else {
+                                                    log.error("Cruise Control getting rebalance proposal failed", rebalanceResult.cause());
+                                                    vertx.cancelTimer(t);
+                                                    p.fail(rebalanceResult.cause());
+                                                }
+                                            });
+                                }
+                            } else {
+                                p.complete(freshClusterRebalance.getStatus());
+                            }
+                        } else {
+                            log.debug("{}: Rebalance resource was deleted, stopping the request time", reconciliation);
+                            vertx.cancelTimer(t);
+                            p.complete();
```

Comment:
In this specific case it's not called because the caller check that the resource is not there anymore so there is no status to update but you are right, I need to check null status in the `updateStatus` for another case, or in general. thanks.